### PR TITLE
libkeyfinder: 2.1 -> 2.2.4

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
https://github.com/ibsh/libKeyFinder#this-repository-is-now-maintained-by-the-mixxx-dj-software-team-please-send-new-pull-requests-and-bug-reports-there

Can someone with knowledge of CMake explain why I see
```
running tests
check flags: SHELL=/nix/store/cwnwyy82wrq53820z6yg7869z8dl5s7g-bash-4.4-p23/bin/bash VERBOSE=y test
Running tests...
/nix/store/mzbylj1zprn18m7h2nsf6vmfafkvz54r-cmake-3.19.3/bin/ctest --force-new-ctest-process
Test project /build/source/build
No tests were found!!!
```
See https://github.com/mixxxdj/libkeyfinder/tree/v2.2.4#testing.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aszlig** [comment]
> Can someone with knowledge of CMake explain why I see
> 
> ```
> running tests
> check flags: SHELL=/nix/store/cwnwyy82wrq53820z6yg7869z8dl5s7g-bash-4.4-p23/bin/bash VERBOSE=y test
> Running tests...
> /nix/store/mzbylj1zprn18m7h2nsf6vmfafkvz54r-cmake-3.19.3/bin/ctest --force-new-ctest-process
> Test project /build/source/build
> No tests were found!!!
> ```

If you look at the lines before, you'll see the actual issue:

```
[100%] Linking CXX executable keyfinder-tests
/build/source/build/tests/keyfinder-tests: error while loading shared libraries: libkeyfinder.so: cannot open shared object file: No such file or directory
/build/source/build/tests/keyfinder-tests: error while loading shared libraries: libkeyfinder.so: cannot open shared object file: No such file or directory
[100%] Built target keyfinder-tests
```

**Comment by dotlambda** [comment]
Do you know how to fix it? I e.g. tried messing around with `CMAKE_LINK_LIBRARY_FLAG` but to no avail.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111633` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>beets</li>
<li>keyfinder</li>
<li>keyfinder-cli</li>
<li>libkeyfinder</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

keyfinder:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern 'keyfinder.0' doesn't match anything in file 'is_KeyFinder.pro'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>

**Comment by aszlig** [comment]
> Do you know how to fix it? I e.g. tried messing around with `CMAKE_LINK_LIBRARY_FLAG` but to no avail.

Ah, I didn't dig deeper than that, but given that upstream runs the `install` step before running tests, we *probably* (note: just guessing) may want to use `installCheckPhase` instead.

**Comment by dotlambda** [comment]
> Ah, I didn't dig deeper than that, but given that upstream runs the `install` step before running tests, we _probably_ (note: just guessing) may want to use `installCheckPhase` instead.

I don't think that's the problem: The tests are built as part of the `buildPhase` in this case.

**Comment by dotlambda** [comment]
From my point of view, this is ready to be merged.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111633` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>beets</li>
<li>keyfinder</li>
<li>keyfinder-cli</li>
<li>libkeyfinder</li>
</ul>
</details>

File: pkgs/development/libraries/libkeyfinder/default.nix [hunk 1]
```diff
@@ -1,37 +1,32 @@
...
   };
 
-  nativeBuildInputs = [ qmake ];
-  buildInputs = [ fftw qtbase ];
-
-  postPatch = ''
-    substituteInPlace LibKeyFinder.pro \
-      --replace "/usr/local" "$out" \
-      --replace "-stdlib=libc++" ""
+  # needed for linking libkeyfinder.so into keyfinder-tests executable
```
Comment (jtojnar) [inline pkgs/development/libraries/libkeyfinder/default.nix:14]: This can be usually resolved using `cmakeFlags = [ "-DCMAKE_SKIP_BUILD_RPATH=OFF" ];`.

Comment (dotlambda) [reply pkgs/development/libraries/libkeyfinder/default.nix:14]: Thanks! Fixed in 85e8220aa70ba637853cb31232a4c12c96760839.

**Pull request merged**